Author: Alexandr Gavriliuc <Alexandr_Gavriliuc@comcast.com>
Date:   Wed Jan 30 17:10:50 2019 +0200
Subject: [PATCH] Revert "[Soup] Correctly set the ignore hosts"
    This reverts commit 13fa6a2cb4bdd11c1f5263fe1202884efb9c6366.
Subject: [PATCH] Revert "[Soup] Use no_proxy env"
    This reverts commit 9450fb4c00fd0ad4d5e9f5d465c81229a24ad2b7.

Signed-off-by: Alexandr Gavriliuc <Alexandr_Gavriliuc@comcast.com>

diff --git a/Source/WebCore/platform/network/soup/SoupNetworkSession.cpp b/Source/WebCore/platform/network/soup/SoupNetworkSession.cpp
index 7f4d58a291f..40b6ec25b00 100644
--- a/Source/WebCore/platform/network/soup/SoupNetworkSession.cpp
+++ b/Source/WebCore/platform/network/soup/SoupNetworkSession.cpp
@@ -261,13 +261,23 @@ void SoupNetworkSession::setProxies(const Vector<WebCore::Proxy>& proxies)
 {
 #if PLATFORM(WPE)
     const char* httpProxy = getenv("http_proxy");
-    const char* noProxy = getenv("no_proxy");
-    if (httpProxy) {
-        GProxyResolver* resolver = g_simple_proxy_resolver_new(httpProxy, (noProxy && strcmp(noProxy, "")) ? g_strsplit(noProxy, ",", -1) : nullptr);
-        g_object_set(m_soupSession.get(), SOUP_SESSION_PROXY_RESOLVER, resolver, nullptr);
+    GProxyResolver* resolver = g_wildcard_proxy_resolver_new(httpProxy);
+
+    GWildcardProxyResolver* w_resolver = G_WILDCARD_PROXY_RESOLVER(resolver);
+
+    GPtrArray* array = g_ptr_array_sized_new(proxies.size());
+    for (size_t i = 0; i < proxies.size(); ++i)
+    {
+        GWildcardProxyResolverProxy* p = g_new(GWildcardProxyResolverProxy, 1);
+        p->pattern = g_strdup(proxies[i].pattern.utf8().data());
+        p->proxy = g_strdup(proxies[i].proxy.utf8().data());
+        g_ptr_array_add(array, p);
     }
-#endif
+    g_wildcard_proxy_resolver_set_proxies(w_resolver, array);
+    g_object_set(m_soupSession.get(), SOUP_SESSION_PROXY_RESOLVER, resolver, nullptr);
+#else
     UNUSED_PARAM(proxies);
+#endif
 }
 
 #if PLATFORM(WPE)
